src/controllers/notification.controller.ts

```diff
@@ -0,0 +1,138 @@
+import mongoose from 'mongoose';
+import { sendNotificationToUser } from '../utils/notification.utils';
+import Notification from '../models/Notification.model';
+
+/**
+ * Retrieves all notifications for a given user.
+ * @param {Request} req - Express Request object containing query parameters.
+ * @param {Response} res - Express Response object containing status and message.
+ * @returns {Promise<Notification[]>} HTTP response indicating success and found notifications.
+ */
+export const getNotifications = async (req: Request, res: Response) => {
+  try {
+    // get userId from query parameters
+    const userId = req.query.userId as string;
+
+    // check if userId is present
+    if (!userId) {
+      return res
+        .status(400)
+        .send({ message: 'Missing userId query parameter.' });
+    }
+
+    // find all notifications for the given user
+    const notifications = await Notification.find({ userId: userId });
+
+    // return found notifications
+    return res.status(200).send({
+      message: 'Found notifications for user!',
+      notifications: notifications,
+    });
+  } catch (error: any) {
+    return res.status(500).send({ message: error?.message });
+  }
+};
+
+/**
+ * Creates and saves a new notification to the database for the specified user.
+ * Does NOT send a notification via expo to a user's device.
+ *
+ * This endpoint is mainly for future admin purposes.
+ *
+ * @param {Request} req - Express request object containing the user data to be stored.
+ * @param {Response} res - Express response object used to send the response back to the client.
+ * @returns {Promise} - a promise resolving to the newly created Notification document from the MongoDB database
+ */
+export const addNotificationsToDatabase = async (
+  req: Request,
+  res: Response
+) => {
+  try {
+    // check if all required fields are present
+    if (
+      !req.body.userIds ||
+      !req.body.title ||
+      !req.body.body ||
+      !req.body.data ||
+      !req.body.notificationType ||
+      req.body.delay === undefined
+    ) {
+      return res.status(500).send({
+        message: 'Missing required fields to create notification for database.',
+      });
+    }
+
+    // check if delay is a positive number
+    if (req.body.delay < 0) {
+      return res
+        .status(500)
+        .send({ message: 'Delay must be a positive number.' });
+    }
+
+    // Javascript will not wait for promises to resolve in a for loop or mapping, so we must collect all the promises
+    const promises = req.body.userIds.map(async (id: string) => {
+      // check if current user id is valid
+      if (mongoose.Types.ObjectId.isValid(id)) {
+        // add notification to database via utils function
+        await sendNotificationToUser(
+          id,
+          req.body.title,
+          req.body.body,
+          req.body.data,
+          req.body.notificationType,
+          req.body.delay
+        );
```

Comment:
will have to refactor this after connecting the `sendNotificationToUser` input params to the `Notification` interface

Reply:
Yes will have to do that